src/tcp_tunnel_tools.js

```diff
@@ -81,26 +69,19 @@ function createTunnel(proxyUrl, targetHost, providedOptions = {}, callback) {
 
 module.exports.createTunnel = createTunnel;
 
-function closeTunnel(serverPath, closeConnections, callback) {
-    const [hostname, port] = serverPath.split(':');
-    if (!hostname) throw new Error('serverPath must contain hostname');
-    if (!port) throw new Error('serverPath must contain port');
-
-    const promise = new Promise((resolve) => {
-        if (!runningServers[port]) return resolve(false);
-        if (!closeConnections) return resolve(true);
-        runningServers[port].connections.forEach((connection) => connection.destroy());
-        resolve(true);
-    })
-        .then((serverExists) => new Promise((resolve) => {
-            if (!serverExists) return resolve(false);
-            runningServers[port].server.close(() => {
-                delete runningServers[port];
-                resolve(true);
-            });
-        }));
+function closeTunnel(address, closeConnections) {
+    if (!runningServers[address]) return false;
```

Comment:
I would make a variable out of `runningServers[address]`. It's used in 3 places and `runningServers[address].connections` and `runningServers[address].server.close()` is a bit too much.